docs/architecture/ErrorReporting.md

```diff
@@ -0,0 +1,122 @@
+# Handling and reporting errors
+
+    WARNING: This document describes future plans for error handling.
+    The UI library does not (yet) implement error handling as described here.
+
+This document describes how errors are handled within the ACS UI library and how they are reported to the surrounding application.
+
+## Error sources
+
+There are two categories of errors, and they are handled differently.
+
+1. ACS backend errors: Errors returned by the REST calls to ACS backend services.
+
+   1. These can be transient or permanent (special case: permission error).
```

Comment:
nit: The special case isn't quite returned by a REST call, this is a browser API call (e.g. no mic permission) - and something that can be resolved by the user at run time
Worthy of its own category? (as these will often by the only time the error will be presented to the web app user)